test/e2e/e2e_test.go

```diff
@@ -258,6 +259,115 @@ float: 123.123
 	})
 }
 
+func TestSchema(t *testing.T) {
+	t.Run("scalar and map defaults", func(t *testing.T) {
+		flags := yttFlags{
+			{"--enable-experiment-schema": ""},
+		}
+		actualOutput := runYtt(t, testInputFiles{"../../examples/schema/config.yml", "../../examples/schema/schema.yml"}, "", flags, nil)
+		expectedOutput := `nothing: null
+string: a string
+bool: false
+int: 0
+float: 0.1
+any: anything
+`
+
+		require.Equal(t, expectedOutput, actualOutput)
+	})
+
+	t.Run("array defaults", func(t *testing.T) {
+		flags := yttFlags{
+			{"--enable-experiment-schema": ""},
+		}
+		actualOutput := runYtt(t, testInputFiles{"../../examples/schema-arrays/config.yml", "../../examples/schema-arrays/schema.yml"}, "", flags, nil)
+		expectedOutput := `jobs: []
+`
+
+		require.Equal(t, expectedOutput, actualOutput)
+	})
+
+	t.Run("--data-value-yaml flag", func(t *testing.T) {
+		flags := yttFlags{
+			{"--data-value-yaml": "nothing=a new string"},
+			{"--data-value-yaml": "string=str"},
+			{"--data-value-yaml": "bool=true"},
+			{"--data-value-yaml": "int=123"},
+			{"--data-value-yaml": "float=123.123"},
+			{"--data-value-yaml": "any=[1,2,4]"},
+			{"--enable-experiment-schema": ""},
+		}
+		actualOutput := runYtt(t, testInputFiles{"../../examples/schema/config.yml", "../../examples/schema/schema.yml"}, "", flags, nil)
+		expectedOutput := `nothing: a new string
+string: str
+bool: true
+int: 123
+float: 123.123
+any:
+- 1
+- 2
+- 4
+`
+
+		require.Equal(t, expectedOutput, actualOutput)
+	})
+
+	t.Run("--data-values-env flag", func(t *testing.T) {
+		flags := yttFlags{
+			{"--data-values-env": "STR_VAL"},
+			{"--data-values-env-yaml": "YAML_VAL"},
+			{"--enable-experiment-schema": ""},
+		}
+		envs := []string{
+			"STR_VAL_nothing=a new string",
+			"YAML_VAL_string=str",
+			"YAML_VAL_bool=true",
+			"YAML_VAL_int=123",
+			"YAML_VAL_float=123.123",
+			"YAML_VAL_any=[1,2,4]",
+		}
+		actualOutput := runYtt(t, testInputFiles{"../../examples/schema/config.yml", "../../examples/schema/schema.yml"}, "", flags, envs)
+		expectedOutput := `nothing: a new string
+string: str
+bool: true
+int: 123
+float: 123.123
+any:
+- 1
+- 2
+- 4
+`
+
+		require.Equal(t, expectedOutput, actualOutput)
+	})
+
+	t.Run("array with data values file", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("defaults missing map items", func(t *testing.T) {
```

**rationale:** what's being exercised here seems to me to be more about the defaulting behavior is happening than that the overriding values are coming from a Data Values file.

Reply:
Removed this test all together in favor of using [`expected.txt`](https://github.com/vmware-tanzu/carvel-ytt/pull/414/files#diff-7484cbd9a3d77b5a1209d32b5dc32729da0515dd6e0cc0ab3e2d74002f772169) to assert on the examples output.